src/engagement_db_to_analysis/code_imputation_functions.py

```diff
@@ -134,26 +150,132 @@ def _impute_age_category(user, messages_traced_data, analysis_dataset_configs):
                 age_category_coding_config.code_scheme, age_category_code, Metadata.get_call_location()
             )
 
-            # Append this age_category_label to the list of labels for this message, and write-back to TracedData.
-            message_labels = message["labels"].copy()
-            message_labels.insert(0, age_category_label.to_dict())
-            message.append_data(
-                {"labels": message_labels},
-                Metadata(user, Metadata.get_call_location(), TimeUtils.utc_now_as_iso_string())
-            )
+            # Inserts this age_category_label to the list of labels for this message, and write-back to TracedData.
+            _insert_label_to_messsage_td(user, message_td, age_category_label)
 
             imputed_labels +=1
 
     log.info(f"Imputed {imputed_labels} age category labels for {age_messages} age messages")
 
 
+def _make_location_code(scheme, clean_value):
+    if clean_value == Codes.NOT_CODED:
+        return scheme.get_code_with_control_code(Codes.NOT_CODED)
+    else:
+        return scheme.get_code_with_match_value(clean_value)
+
+
+def _impute_kenya_location_codes(user, messages_traced_data, analysis_dataset_configs):
+    """
+    Imputes Kenya location labels for location dataset messages.
+
+    :param user: Identifier of user running the pipeline.
+    :type user: str
+    :param messages_traced_data: Messages TracedData objects to impute age_category.
+    :type messages_traced_data: list of TracedData
+    :param analysis_dataset_configs: Analysis dataset configuration in pipeline configuration module.
+    :type analysis_dataset_configs: pipeline_config.analysis_configs.dataset_configurations
+    """
+
+    log.info(f"Imputing Kenya location labels for location messages...")
+
+    # Get the coding configurations for constituency and county analysis datasets
```

Comment:
I suggest Line 183 - 209 be in a function that returns `county_coding_config, constituency_coding_config and location_engagement_db_datasets`. You can return a tuple so that you can unpack it later in _impute_kenya_location_codes i.e 
`county_coding_config, constituency_coding_config, location_engagement_db_datasets = get_ke_location_cc(analysis_dataset_configs)`